catalog/transformer/catalog-transformer-zip/src/test/java/org/codice/ddf/catalog/transformer/zip/TestZipCompression.java

```diff
@@ -38,135 +36,234 @@
+
+    private static final List<String> METACARD_RESULT_LIST_NO_CONTENT = Arrays.asList(METACARD_1,
+            METACARD_2,
+            METACARD_3);
+
+    private static final List<String> METACARD_RESULT_LIST_WITH_CONTENT = Arrays.asList(METACARD_1,
+            METACARD_2,
+            METACARD_3,
+            METACARD_3_CONTENT);
+
+    private static final List<String> METACARD_RESULT_LIST_WITH_CONTENT_AND_DERIVED_RESOURCES =
+            Arrays.asList(METACARD_1,
+                    METACARD_2,
+                    METACARD_3,
+                    METACARD_3_CONTENT,
+                    METACARD_3_DERIVED_CONTENT);
 
-    File file = new File(ZIP_FILE_PATH);
+    private ZipCompression zipCompression;
+
+    private SourceResponse sourceResponse;
+
+    private Map<String, Serializable> filePathArgument;
+
+    private CatalogFramework catalogFramework;
 
     private InputStream resourceFileStream;
 
     @Before
     public void setUp() throws Exception {
         zipCompression = new ZipCompression();
-        sourceResponse = generateMetacardList("ddf.distribution", false, Arrays.asList("id1",
-                "id2"));
+        sourceResponse = createSourceResponseWithURISchemes(null, null);
         filePathArgument = new HashMap<>();
-        filePathArgument.put("filePath", ZIP_FILE_PATH);
+        filePathArgument.put("filePath",
+                temporaryFolder.getRoot()
+                        .getAbsolutePath() + File.separator + "signed.zip");
         catalogFramework = mock(CatalogFramework.class);
 
         Resource resource = mock(Resource.class);
         resourceFileStream = new FileInputStream(new File(LOCAL_RESOURCE_PATH));
         when(resource.getName()).thenReturn(LOCAL_RESOURCE_FILENAME);
         when(resource.getInputStream()).thenReturn(resourceFileStream);
         ResourceResponse resourceResponse = new ResourceResponseImpl(resource);
-        when(catalogFramework.getLocalResource(any(ResourceRequest.class))).thenReturn(
+        when(catalogFramework.getLocalResource(any(ResourceRequestById.class))).thenReturn(
                 resourceResponse);
         zipCompression.setCatalogFramework(catalogFramework);
+    }
 
+    @Test
+    public void testGetCatalogFramework() {
+        assertThat(catalogFramework, is(zipCompression.getCatalogFramework()));
     }
 
     @Test(expected = CatalogTransformerException.class)
     public void testCompressionNullArguments() throws Exception {
         BinaryContent binaryContent = zipCompression.transform(sourceResponse, null);
         assertThat(binaryContent, nullValue());
-        Files.deleteIfExists(file.toPath());
+    }
+
+    @Test(expected = CatalogTransformerException.class)
+    public void testCompressionEmptyArguments() throws Exception {
+        BinaryContent binaryContent = zipCompression.transform(sourceResponse, new HashMap<>());
+        assertThat(binaryContent, nullValue());
```

Comment:
Shouldn't need `assertThat(binaryContent, nullValue())` in any of these tests that expect exceptions.